Mathlib/LinearAlgebra/Eigenspace/Semisimple.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2024 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.LinearAlgebra.Eigenspace.Basic
+import Mathlib.LinearAlgebra.Semisimple
+
+/-!
+# Eigenspaces of semisimple linear endomorphisms
+
+This file contains basic results relevant to the study of eigenspaces of semisimple linear
+endomorphisms.
+
+## Main definitions / results
+
+ * `Module.End.IsSemisimple.generalizedEigenspace_eq_eigenspace`: for a semisimple endomorphism,
+   a generalized eigenspace is an eigenspace.
+
+-/
+
+open Function Set
+
+namespace Module.End
+
+variable {R M : Type*} [CommRing R] [AddCommGroup M] [Module R M] {f g : End R M}
+
+lemma apply_eq_of_mem_genEigenspace_of_comm_of_isSemisimple_of_isNilpotent_sub
```

Comment:
Should we just rename `generalizedEigenspace` to `genEigenspace` globally? That name is wicked long.

Reply:
I agree. I can do this if you wish (either before or after this PR). I note however that it might be more work than a simple search-and-replace if we wish to put `@[deprecated] alias ` notes in place.

Reply:
Let's do it later.